rclcpp/include/rclcpp/experimental/intra_process_manager.hpp

```diff
@@ -324,8 +301,39 @@ class IntraProcessManager
 
   struct SplittedSubscriptions
   {
-    std::vector<uint64_t> take_shared_subscriptions;
-    std::vector<uint64_t> take_ownership_subscriptions;
+    enum
+    {
+      IndexSharedTyped = 0, IndexSharedSerialized = 1,
+      IndexOwnershipTyped = 2, IndexOwnershipSerialized = 3,
+      IndexNum = 4
+    };
+
+    /// get the index for "take_subscriptions" depending on shared/serialized
+    constexpr static auto
```

Comment:
```suggestion
    constexpr static size_t
```

I had to change it to `size_t` in order to make it work on OSX. Otherwise I got:
```
intra_process_manager.hpp:325:67: error: function 'get_index' with deduced return type cannot be used before it is defined
    constexpr static auto ownership_same = SplittedSubscriptions::get_index(
                                                                  ^
/Users/karsten/workspace/ros2/ros2_master/src/ros2/rclcpp/rclcpp/include/rclcpp/experimental/intra_process_manager.hpp:313:5: note: 'get_index' declared here
    get_index(const bool is_shared, const bool is_serialized)
```

Reply:
More precisely, this should return an `Index`.